desktop/apps/auction2/components/DOM.js

```diff
@@ -0,0 +1,115 @@
+import AddToCalendar from 'desktop/components/add_to_calendar/index.coffee'
+import ClockView from 'desktop/components/clock/view.coffee'
+import ConfirmRegistrationModal from 'desktop/components/credit_card/client/confirm_registration.coffee'
+import MyActiveBids from 'desktop/components/my_active_bids/view.coffee'
+import PropTypes from 'prop-types'
+import { Component } from 'react'
+import infiniteScroll from 'desktop/apps/auction2/utils/infiniteScroll'
+import mediator from 'desktop/lib/mediator.coffee'
+import scrollToTop from 'desktop/apps/auction2/utils/scrollToTop'
+
+export default class DOM extends Component {
+  static propTypes = {
+    auction: PropTypes.object.isRequired,
+    children: PropTypes.node.isRequired,
+    store: PropTypes.object.isRequired,
+    user: PropTypes.object.isRequired
+  }
+
+  // Backbone views
+  clockView = null
+  addToCalendarView = null
+  myActiveBidsView = null
+
+  // Selectors
+  $ = null
+  $body = null
+  $registerBtn = null
+
+  componentDidMount () {
```

Comment:
Just an FYI for those reviewing: Component lifecycle methods are not triggered in any way during server-side renders, and working within them is the appropriate way for dealing with cross-boundary concerns. While everything passes through the render cycle on the server, client-specific code is safe and is only "activated" when React reconciles the two view representations on the client during rehydration.